### ISSUE TITLE
Fix wrong page title and missing head tags when page is rendered server-side

### ISSUE THREAD

**Issue by stephanethomas** [Automattic/wp-calypso#18919]
We should fix server-side rendering which no longer retrieves the right cached data, which leads to serving a page with the generic page title, and no meta or link tags. Note this issue doesn't happen when the page is taken over and re-rendered client-side. This affects search engines that now display a wrong title, and possibly duplicate content.

1. Run `git checkout master`, and start your server
2. View the **source** of the [`Login` page](http://calypso.localhost:3000/log-in)
3. Assert that the title is `Log In — WordPress.com`, and that there is a `canonical` `link` tag:
```html
<html lang="en" dir="ltr">
  <head>
    <title>Log In — WordPress.com</title>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=Edge">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="format-detection" content="telephone=no">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="referrer" content="origin">
    <meta property="og:site_name" content="WordPress.com">
    <link rel="canonical" href="https://wordpress.com/login">
```
4. Reload the source view, and notice that the title is wrong now, and that the tag disappeared:
```html
<html lang="en" dir="ltr">
  <head>
    <title>WordPress.com</title>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=Edge">
    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">
    <meta name="format-detection" content="telephone=no">
    <meta name="mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="referrer" content="origin">
    <meta property="og:site_name" content="WordPress.com">
```

The OAuth flow is also affected:

1. Open the Akismet [`Home` page](https://akismet.com/)
2. Click the `Sign In` button in the upper right corner
3. Assert that you are redirected to the `Login` page
4. View the source of this page, and notice the wrong title and missing tag again

**Comment by stephanethomas** [comment]
This happens because `getCacheKey()` is deemed to accept [a `context` object](https://github.com/Automattic/wp-calypso/blob/2a61468abb79f6b16d4c5cb427f202dc68f4004b/server/isomorphic-routing/index.js#L98) with at least the `query` and `pathname` property, as well as an optional `cacheQueryKeys`. However, we pass it a [`request` object](https://github.com/Automattic/wp-calypso/blob/b945192a093cc0be8137e1b95f2f46353f43cac6/server/pages/index.js#L146) server-side which only contains the `query` property. It does contains a `_parsedUrl` prop with `query` and `pathname` though:

```
  [...]
  _parsedUrl:
   Url {
     protocol: null,
     slashes: null,
     auth: null,
     host: null,
     port: null,
     hostname: null,
     hash: null,
     search: null,
     query: null,
     pathname: '/log-in',
     path: '/log-in',
     href: '/log-in',
     _raw: '/log-in' },
  params:
   { twoFactorAuthType: undefined,
     lang: undefined,
     flow: undefined,
     socialService: undefined },
  query: {},
  [...]
```

However we still miss the `cacheQueryKeys` property. It's because it is [set in controllers](https://github.com/Automattic/wp-calypso/blob/75833401c19b407e30b282dfe1283a45554660c6/client/login/controller.js#L24), which happens too late in the processing of the request, after the call to `getCacheKey()`.

In a nutshell, the page title and tags are [correctly cached](https://github.com/Automattic/wp-calypso/blob/36d99d0638279f0a6ba28366cabbe85a4f0315f4/server/render/index.js#L107) on the first page load with the right cache key. However, we can't retrieve this data on subsequent calls because of an incorrect cache key.
